src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildInfo.java

```diff
@@ -101,9 +101,20 @@ public void addBuilder(BuildBotData update) {
     builders.add(update);
   }
 
-  public boolean equals(BuildInfo other) {
-    return this.commitHash == other.commitHash && this.timestamp.equals(other.timestamp) &&
-           this.branch == other.branch && this.builders.equals(other.builders);
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (o == null) {
+      return false;
+    }
+    if (!(o instanceof BuildInfo)) {
+      return false;
+    }
+    BuildInfo other = (BuildInfo) o;
+    return commitHash.equals(other.commitHash) && timestamp.equals(other.timestamp) &&
+      branch.equals(other.branch) && builders.equals(other.builders);
```

Comment:
nit: perhaps indent line 117 to start at the same column as `commitHash.equals...`.

Reply:
Agree! Thanks for noticing it